bitcoin/src/blockdata/script.rs

```diff
@@ -170,14 +170,54 @@ pub enum Error {
     Serialization
 }
 
+// If bitcoinonsensus-std is off but bitcoinconsensus is present we patch the error type to
+// implement `std::error::Error`.
+#[cfg(all(feature = "std", feature = "bitcoinconsensus", not(feature = "bitcoinconsensus-std")))]
```

Comment:
Why don't we just use
```
compile_error!("In order to use the features bitcoinconsensus and std please enable bitcoinconsensus-std as well");
```
So we fail to compile if users don't enable `bitcoinconsensus-std`. Then we can do away with all the error wrapping.

Reply:
Because we may have one crate in a dep tree that uses `std`, and another one that uses `bitcoinconsensus` but not `std`, and no individual crate can be held responsible for enabling `bitcoinconsensus-std`.

Ultimately I feel like this problem is a cargo bug (or design flaw) rather than something we should have to deal with ... but it is what it is.

Reply:
I think I understand the problem, my idea is just instead of wrapping the error when the problem arises, fail to build so that the responsibility of enabling things correctly is forced onto `rust-bitcoin` (and yes I agree that enabling features in optional dependencies seems broken in general).

Reply:
We really do not want to "fail to build" when we're included in certain shapes of dependency tree.

Reply:
Oh of course, that is the whole reason for starting this work - face palm

Reply:
> I feel like this problem is a cargo bug (or design flaw)

Yes, it is and it's fixed in 1.65 - yep, to be released soon.

>  fail to build so that the responsibility of enabling things correctly is forced onto `rust-bitcoin`

That will not force responsibility on `rust-bitcoin` but on whichever crate happened to include two dependencies which themselves include `rust-bitcoin`.